### PR TITLE
Add merge_response templating

### PR DESCRIPTION
**Pull request by gjohansson-ST**
## Proposed change

Adds `merge_response` to HA templating

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/114204
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **Documentation**
	- Enhanced the templating documentation with a new section on `merge_response` functionality, detailing parameters `value`, `sort_by`, and `selected_key`, along with practical examples to guide users in merging action responses effectively.
	- Added comprehensive examples in both YAML and JSON formats to illustrate the merging of multiple calendar event responses.

### CODE REVIEW

File: source/_docs/configuration/templating.markdown [hunk 1]
```diff
@@ -1148,6 +1148,31 @@ See: [Python regular expression operations](https://docs.python.org/3/library/re
...
+
+| Variable     | Description                                            |
+| ------------ | ----------------------------------                     |
+| `value`      | The incoming value (action response).                  |
+| `sort_by`    | Sort a response dictionary by this key.                |
+| `single_key` | Return a list with information from a single key only. |
+
+{% note %}
+
+`sort_by` and `single_key` has only use if the action call is returning dictionaries
```
Comment (gjohansson-ST) [inline source/_docs/configuration/templating.markdown]:
```suggestion
`sort_by` and `single_key` have only use if the action call is returning dictionaries, for example `weather` and `calendar` action responses.
```

**Review by emontnemery — DISMISSED** [review]
Let's add back an example on how to sort, but using Jinja's `sort` filter instead of the removed sort parameter.
For example, to sort the calendar events by start time:
```
merge_response(response_variable) | sort(attribute='start')
```

File: source/_docs/configuration/templating.markdown [hunk 2]
```diff
@@ -1148,6 +1148,96 @@ See: [Python regular expression operations](https://docs.python.org/3/library/re
 - Filter `value | regex_findall(find='', ignorecase=False)` will find all regex matches of the find expression in `value` and return the array of matches.
 - Filter `value | regex_findall_index(find='', index=0, ignorecase=False)` will do the same as `regex_findall` and return the match at index.
 
+## Merge action responses
+
+Using action responses we can collect information from various entities at the same time.
+Using the `merge_response` template we can merge several responses into one list.
```
Comment (emontnemery) [inline source/_docs/configuration/templating.markdown:1154]: We should explain the `entity_id` will be added to the template output, and that the function will fail if there's already a key `entity_id`.

Also, let's add an example for the case where the response is a scalar.

File: source/_docs/configuration/templating.markdown [hunk 3]
```diff
@@ -1148,6 +1148,178 @@ See: [Python regular expression operations](https://docs.python.org/3/library/re
...
+
+| Variable       | Description                                     |
+| -------------- | ----------------------------------              |
+| `value`        | The incoming value (must be an action response). |
+
+`entity_id` key is appended to each dictionary within the template output list as a reference of origin. If the input dictionary already contains an `entity_id` key the template will fail.
```
Comment (c0ffeeca7) [inline source/_docs/configuration/templating.markdown]:
```suggestion
The `entity_id` key is appended to each dictionary within the template output list as a reference of origin. If the input dictionary already contains an `entity_id` key, the template will fail.
```

File: source/_docs/configuration/templating.markdown [hunk 4]
```diff
@@ -1148,6 +1148,178 @@ See: [Python regular expression operations](https://docs.python.org/3/library/re
...
+
+`value_key` key is appended to each dictionary within the template output list as a reference of origin if the original service call was providing a list of dictionaries, as example `calendar.get_events` or `weather.get_forecasts`.
```
Comment (c0ffeeca7) [inline source/_docs/configuration/templating.markdown]:
```suggestion
The `value_key` key is appended to each dictionary within the template output list as a reference of origin if the original service call was providing a list of dictionaries, for example, `calendar.get_events` or `weather.get_forecasts`.
```

File: source/_docs/configuration/templating.markdown [hunk 5]
```diff
@@ -1148,6 +1148,178 @@ See: [Python regular expression operations](https://docs.python.org/3/library/re
...
+
+Examples of these two keys can can be seen in [example merge calendar action response](#example-merge-calendar-action-response) template output.
```
Comment (c0ffeeca7) [inline source/_docs/configuration/templating.markdown]:
```suggestion
Examples of these two keys can be seen in [example merge calendar action response](#example-merge-calendar-action-response) template output.
```

File: source/_docs/configuration/templating.markdown [hunk 6]
```diff
@@ -1148,6 +1148,178 @@ See: [Python regular expression operations](https://docs.python.org/3/library/re
...
+
+{% set combined_forecast = merge_response(response) %}
+{{ combined_forecast[0].precipitation | float(0) | round(1) }}
+
+{% endraw %}
+```
+
+### Example how to sort
+
+Sorting the dictionaries within the list based on a specific key can be done directly by using the jinja's `sort` filter.
```
Comment (c0ffeeca7) [inline source/_docs/configuration/templating.markdown]:
```suggestion
Sorting the dictionaries within the list based on a specific key can be done directly by using Jinja's `sort` filter.
```

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @gjohansson-ST :+1:

**Pull request merged**